File: pet-clinic-java/src/main/java/com/github/rskupnik/petclinic/model/Pet.java
package com.github.rskupnik.petclinic.model;

import javax.persistence.*;

@Entity
public class Pet {

    @Id
    @GeneratedValue
    private Long id;
    private String name;

    @ManyToOne
    @JoinColumn(name = "ownerId", nullable = false)
    private Customer owner;

    protected Pet() {

    }

    public Pet(String name) {
        this.name = name;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Customer getOwner() {
        return owner;
    }

    public void setOwner(Customer owner) {
        this.owner = owner;
    }

    @Override
    public String toString() {
        return name;
    }
}

File: pet-clinic-kotlin/src/main/kotlin/com/github/rskupnik/petclinic/model/Pet.kt
package com.github.rskupnik.petclinic.model

import javax.persistence.*

// Implementation using a regular class, mimicking regular Java

@Entity
class Pet {

    // We need to specify the default constructor and the parameterized constructor explicitly
    constructor() {

    }

    constructor(name: String) {
        this.name = name
    }

    @Id
    @GeneratedValue
    var id: Long = 0

    var name: String = ""

    @ManyToOne
    @JoinColumn(name = "ownerId", nullable = false)
    var owner: Customer? = null

    override fun toString(): String = "$name"
}